lib/index.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+const request = require('request-promise-native');
+const cache = require('./cache');
+
+const defaultInt = 1000;
+
+module.exports = endpoints => {
+
+  return (req, res, next) => {
+    if (endpoints && endpoints.length) {
+
+      let requests = endpoints.reduce((memo, option) => {
+        const url = option.url;
+        const interval = parseInt(option.interval, 10) || defaultInt;
+        const currTime = Date.now();
+        const futureTime = currTime + interval;
+        const savedTime = cache.get(url);
+
+        // Is this the first time or has the interval expired
+        if (isNaN(savedTime) || savedTime < currTime) {
+          cache.set(url, futureTime);
+          memo.push(request(url).promise());
+        }
+        return memo;
+
+      }, []);
+
+      // Process all pending requests
+      return Promise.all(requests)
+        .then(resolved => {
+          for (var i = 0; i < resolved.length; i++) {
+            let response;
+            try {
+              response = JSON.parse(resolved[i]);
+            } catch (e) {
+              req.log('debug', response, e);
+            }
+
+            if (response) {
+              const status = response.statusCode || response.status;
+              if (status.startsWith(5)) {
```

Comment:
why not just check for a successful 200 there are more errors than 500s that you could get back. This is flakey.